android/quest/src/main/java/org/smartregister/fhircore/quest/ui/report/measure/worker/MeasureReportWorker.kt

```diff
@@ -0,0 +1,174 @@
+    fhirEngine.search<Measure> {}.forEach {
+      monthList.forEachIndexed { index, date ->
+        val startDateFormatted = date.firstDayOfMonth().formatDate(SDF_YYYY_MM_DD)
+        val endDateFormatted = date.lastDayOfMonth().formatDate(SDF_YYYY_MM_DD)
+        if (alreadyGeneratedMeasureReports(fhirEngine, startDateFormatted, endDateFormatted, it.url)
+            .isEmpty()
+        ) {
+          evaluatePopulationMeasure(it.url, startDateFormatted, endDateFormatted)
+        } else {
+          if (index == 0 && lastDayOfMonth(endDateFormatted.parseDate(SDF_YYYY_MM_DD)))
+            evaluatePopulationMeasure(it.url, startDateFormatted, endDateFormatted)
+        }
+      }
+    }
+
+    return Result.success()
+  }
+
+  private fun lastDayOfMonth(date: Date?): Boolean = date?.before(today().lastDayOfMonth()) == true
+
+  private suspend fun evaluatePopulationMeasure(
+    measureUrl: String,
+    startDateFormatted: String,
+    endDateFormatted: String
+  ) {
+    withContext(dispatcherProvider.io()) {
+      fhirEngine.loadCqlLibraryBundle(fhirOperator, measureUrl)
+    }
+    val measureReport: MeasureReport? =
+      withContext(dispatcherProvider.io()) {
+        try {
+          fhirOperator.evaluateMeasure(
+            measureUrl = measureUrl,
+            start = startDateFormatted,
+            end = endDateFormatted,
+            reportType = MeasureReportViewModel.POPULATION,
+            subject = null,
+            practitioner = null
+            /* TODO DO NOT pass this id to MeasureProcessor as this is treated as subject if subject is null.
+            practitionerId?.asReference(ResourceType.Practitioner)?.reference*/ ,
+            lastReceivedOn = null // Non-null value not supported yet
+          )
+        } catch (exception: IllegalArgumentException) {
+          Timber.e(exception)
+          null
+        }
+      }
+    if (measureReport != null) {
+      val result =
+        alreadyGeneratedMeasureReports(fhirEngine, startDateFormatted, endDateFormatted, measureUrl)
+      if (result.isNotEmpty()) defaultRepository.delete(result.last())
+      defaultRepository.addOrUpdate(resource = measureReport)
+    }
+  }
+
+  /** @return list of months within current date and starting date of campaign */
+  fun getMonthRangeList(): List<Date> {
+    Timber.w("appConfig$applicationConfiguration")
+    val yearMonths = mutableListOf<Date>()
+    val endDate = Calendar.getInstance().time.formatDate(SDF_YYYY_MM_DD).parseDate(SDF_YYYY_MM_DD)
+    var lastDate = endDate?.firstDayOfMonth()
+    while (lastDate?.after(applicationConfiguration.registerDate.parseDate(SDF_YYYY_MM_DD)) ==
+      true) {
+      yearMonths.add(lastDate)
+      lastDate = lastDate.plusMonths(-1)
+    }
+    return yearMonths.toList()
+  }
+  companion object {
+    private const val WORK_ID = "fhirMeasureReportWorker"
+
+    fun scheduleMeasureReportWorker(workManager: WorkManager, hours: Int, minutes: Int) {
+      // trigger at 8:30am
```

Comment:
This comment is confusing. Will this always be triggered at 8.30am?

Reply:
Removing it.